packages/rolldown/src/plugin/plugin-context.ts

```diff
@@ -188,10 +190,10 @@ export class PluginContext extends MinimalPluginContext {
     this.context.addWatchFile(id)
   }
 
-  /**
-   * @deprecated This rollup API won't be supported by rolldown. Using this API will cause runtime error.
-   */
-  public parse(_input: string, _options?: any): any {
-    unsupported('`PluginContext#parse` is not supported by rolldown.')
+  public parse(
+    input: string,
+    options?: ParserOptions | undefined | null,
+  ): Program {
+    return parseAst('test.js', input, options)
```

Comment:
Is this `'test.js'` supposed to be hardcoded?

Reply:
This needs to be harded coded (or any meaningful filename) due to oxc parseAst, however `rollup` parseAst did not need filename

Reply:
In case of errors or logs, will `'test.js'` be output anywhere?

Reply:
Seems no, see https://github.com/rolldown/rolldown/pull/3586/files#diff-f545b0a3b1d006ae8bab76110e18fa81c29297a4f76f87c6566b8cac5a6611eaL21